moveit_ros/planning_interface/planning_scene_interface/src/planning_scene_interface.cpp

```diff
@@ -80,6 +87,10 @@ class PlanningSceneInterface::PlanningSceneInterfaceImpl
     moveit_msgs::GetPlanningScene::Response response;
     std::vector<std::string> result;
     request.components.components = request.components.WORLD_OBJECT_NAMES;
+
+    if (!planning_scene_service_.isValid())
+      connectGetPlanningSceneService();
+
     if (!planning_scene_service_.call(request, response))
```

Comment:
Obviously, you need to ensure a valid connection before each call. Please encapsulate these two steps within new methods `get_planning_scene_service_call` and `apply_planning_scene_service_call`, which will clarify the interdependency.

Reply:
Ok, I would call them `getPlanningSceneServiceCall` and `applyPlanningSceneServiceCall` for consistency with the other class methods, is that ok with you?

Reply:
Sure.